File: java/v1/src/main/java/com/api/v1/borrower/services/BorrowerSelfRegistrationService.java
package com.api.v1.borrower.services;

import com.api.v1.borrower.domain.Borrower;
import com.api.v1.borrower.helpers.BorrowerResponseDto;
import com.api.v1.borrower.helpers.NewBorrowerRequestDto;

import jakarta.validation.Valid;
import reactor.core.publisher.Mono;

public interface BorrowerSelfRegistrationService {

    Mono<BorrowerResponseDto> selfRegister(@Valid NewBorrowerRequestDto request);

}

File: koltlin/v1/src/main/kotlin/com/api/v1/borrower/services/BorrowerSelfRegistrationService.kt
package com.api.v1.borrower.services

import com.api.v1.borrower.helpers.BorrowerResponseDto
import com.api.v1.borrower.helpers.NewBorrowerRequestDto
import reactor.core.publisher.Mono

interface BorrowerSelfRegistrationService {

    fun selfRegister(request: NewBorrowerRequestDto): Mono<BorrowerResponseDto>

}